### PR TITLE
(APS-305) Show task deadlines

### PR DESCRIPTION
**Pull request by pezholio**
This alters the transformer to show the calculated task deadlines as introduced in #1538, it also alters the `/tasks` endpoint to sort by a task's `dueAt` value, because now tasks have different due dates, it's no longer going to be sufficient to sort by a task's created date to "fake" sorting by due date.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/TasksTest.kt [hunk 1]
```diff
@@ -896,8 +900,9 @@ class TasksTest : IntegrationTestBase() {
       }
     }
 
-    @Test
-    fun `Get all reallocatable tasks returns 200 when no type retains original sort order`() {
+    @ParameterizedTest
+    @CsvSource(value = ["createdAt,asc", "createdAt,desc", "dueAt,asc", "dueAt,desc"])
```
Comment (davidatkinsuk) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/TasksTest.kt:904]: this is cool, much better than defining a function on some Object elsewhere

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/TaskTransformer.kt [hunk 2]
```diff
@@ -117,4 +119,9 @@ class TaskTransformer(
       userTransformer.transformJpaToApi(userEntity, ServiceName.approvedPremises) as ApprovedPremisesUser
     }
   }
+
+  // Use the sure operator here as entities will definitely have a `dueAt` value by the time they're surfaced as tasks
```
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/TaskTransformer.kt:123]: presumably this has all been backfilled in prod/pre-prod before we merge this (or does the code ensure it's populated on retrieval?)

Comment (pezholio) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/TaskTransformer.kt:123]: Yeah, it's done in preprod, but I've yet to do it in prod. Will do that before merging this

**Review by davidatkinsuk — APPROVED** [review]
LGTM 👍

**Pull request merged**